bin/setup.csh

```diff
@@ -0,0 +1,46 @@
+# setup lsstsw environment
+#
+# source this file from your ~/.cshrc
+#
+# possibly blank, but can't be parsed as $_
+set cmd=($_)
+if ( $?0 ) then
+    set source=$0
+else if ( $#cmd >= 3 ) then
+    set source=`pwd`/${cmd[2]}
+else if (-f /usr/sbin/lsof ) then
+    set source=`/usr/sbin/lsof +p $$ | grep -oE /.\*setup.csh`
```

Comment:
Maybe so, but `$LSSTSW/bin/setup.sh` is already designed as a login script. If you want to change whether it's a login or call-only script, we'd also have to modify the bash version of the script so that it agrees.

As for your last example, `$_` only contains useful information if you've directly sourced the script from the command line. It's blank if the script was sourced from another script. :frowning: Though I agree that code handles the relative- vs. absolute-path question better and I'll change the parsing accordingly.